mantidimaging/gui/windows/spectrum_viewer/test/presenter_test.py

```diff
@@ -285,18 +289,34 @@ def test_WHEN_rits_roi_clicked_THEN_rois_not_updated(self):
     def test_WHEN_ROI_renamed_THEN_roi_renamed(self):
         rois = ["all", "roi", "roi_1"]
         self.view.spectrum_widget.roi_dict = {roi: mock.Mock() for roi in rois}
-        self.presenter.model._roi_ranges = {roi: mock.Mock() for roi in rois}
-        self.view.spectrum_widget.rename_roi = mock.Mock()
+        self.view.spectrum_widget.rois = {roi: mock.Mock() for roi in rois}
+        self.view.spectrum_widget.rename_roi = mock.Mock(
+            side_effect=lambda old_name, new_name: self.view.spectrum_widget.rois.update(
+                {new_name: self.view.spectrum_widget.rois.pop(old_name)}))
+
         self.presenter.rename_roi("roi_1", "new_name")
         self.view.spectrum_widget.rename_roi.assert_called_once_with("roi_1", "new_name")
-        self.assertIn("new_name", self.presenter.model._roi_ranges)
-        self.assertNotIn("roi_1", self.presenter.model._roi_ranges)
+        self.assertIn("new_name", self.view.spectrum_widget.rois)
+        self.assertNotIn("roi_1", self.view.spectrum_widget.rois)
 
-    @parameterized.expand([("all", ), ("roi", )])
-    def test_WHEN_ROI_renamed_to_existing_name_THEN_runtimeerror(self, name):
+    def test_WHEN_invalid_ROI_renamed_THEN_error_raised(self):
         rois = ["all", "roi", "roi_1"]
         self.view.spectrum_widget.roi_dict = {roi: mock.Mock() for roi in rois}
         self.presenter.model._roi_ranges = {roi: mock.Mock() for roi in rois}
+        self.view.spectrum_widget.rename_roi = mock.Mock(side_effect=KeyError("Invalid ROI"))
+        self.view.spectrum_widget.rois = {roi: mock.Mock() for roi in rois}
+        with self.assertRaises(KeyError):
+            self.presenter.rename_roi("invalid_roi", "new_name")
+
+    @parameterized.expand([("all", ), ("roi", )])
+    def test_WHEN_ROI_renamed_to_existing_name_THEN_keyerror(self, name):
+        rois = ["all", "roi", "roi_1"]
+        self.view.spectrum_widget.roi_dict = {roi: mock.Mock() for roi in rois}
+        self.view.spectrum_widget.rois = {roi: mock.Mock() for roi in rois}
+
+        self.view.spectrum_widget.rename_roi = mock.Mock(side_effect=lambda old_name, new_name: (
+            (_ for _ in ()).throw(KeyError(f"Cannot rename to existing ROI: {new_name}"))
+            if new_name in self.view.spectrum_widget.rois else None))
         with self.assertRaises(KeyError):
             self.presenter.rename_roi("roi", name)
```

Comment:
These are also testing the test. Right now `spectrum_widget.rename_roi` never raises an exception, so this is testing a response that doesn't happen.

Better would be to implement the checks in `spectrum_widget.rename_roi` and test them in the test for that. e.g. based on the tests removed from the model.